knowledge-base/adr/0024-url-encoding.md

```diff
@@ -0,0 +1,191 @@
+One can use an IDN converter so `www.häuser.de` can be converted to `www.xn--huser-gra.de`.
+As the host is usually configured in the destination, the SDK will not make any adjustment.
+
+### Path
+
+#### OData
+
+Below is one example about how the path should look like in terms of OData
+
+```
+/servicePath/Entity('key')/to_MultiLink('key')/to_SingleLink
+```
+
+- service path: `/servicePath` :thumbsdown:
+- entity path: `/Entity(KeyName='value')` :thumbsup:
+- navigation property path: `/to_MultiLink(KeyName='value')/to_SingleLink` :thumbsup:
+
+##### Service Path :thumbsdown:
+
+We assume the service path has not special characters that need to be encoded.
+If needed, the user can easily configure the service mapping file.
+
+##### Entity Path :thumbsup:
+
+The entity path consists of the following parts:
+
+- an entity name: `Entity` :thumbsdown:
+- entity key names: `KeyName` :thumbsdown:
+- values of the entity key: `'value'` :thumbsup:
+
+We assume the "entity names" and "entity key names" provided by the metadata do not have to be encoded.
+If needed, based on the feedback, we can either do it during generation or ask the possibility to adjust the metadata.
+
+On the other side, the "values of the entity key" of an entity is provided by the user, which might contain special characters.
+One of the users uses `/` as part of a key.
+Therefore, we have to encode the "values of the entity key".
+
+##### Navigation Property Path :thumbsup:
+
+Similar to entity path, only the "values of the navigation property key" are planned to be encoded.
+
+#### OpenAPI
+
+The path pattern should look like the example below:
+
+```
+"/entities/{entityId}"
+```
+
+- path: `/entities` :thumbsdown:
+- path parameter: `{entityId}` :thumbsup:
+
+Similar to OData, only user input is considered when encoding, which is the "path parameters"
+
+### Query
+
+#### OData
+
+OData allows you to use the following query options:
+
+- `$format=json` :thumbsdown:
+- `$top=1` :thumbsdown:
+- `$skip=1` :thumbsdown:
+- `$orderby=StringProperty` :thumbsdown:
+- `$select=StringProperty` :thumbsdown:
+- `$expand=to_MultiLink` :thumbsup:
+- `$filter=StringProperty eq 'string'` :thumbsup:
+- `$batch` :thumbsdown:
+- `$value` :thumbsdown:
+
+`$filter` is the only one query option that contains user input.
+One spacial case is the `$expand`.
+Normal `$expand` has the same structure as `$select`, while for deep `$expand`, it depends on the inner query option.
+
+#### OpenAPI
+
+The value of the query parameters should be encoded. :thumbsup:
+
+### Which parts are encoded by the SDK?
+
+To wrap up, the SDK will only encode the parts below:
+
+- The values of path parameters (the get by key structure of OData is called path parameter in OpenAPI)
+- The values of query parameters
+
+## Decision
+
+Option B is the winner, as Option A does not fit our requirement.
```

Comment:
Could you perhaps change the order:
- Alternatives
- Decision->encode parts not full URL
- Explain which parts are encoded by SDK